### ISSUE TITLE
NOK: Minting limit exceed on polygon Retrying

### ISSUE THREAD

**Issue by cagri27** [HashLips/hashlips_art_engine#632]
I did everyting right but when i go to terminal and run: node utils/nftport/mint.js 
at first it is going well and minting NFTs. but after minting 140 NFTs it says; NOK: Minting limit exceed on polygon Retrying

How can i solve this issue?

**Comment by bolshoytoster** [comment]
As said [here](https://www.reddit.com/r/NFT/comments/sdq2p9/help_with_nft_minting_polygon/huel5ng/), NFT collections are typically minted in one transaction, so wherever you got this script (presumably [this](https://github.com/codeSTACKr/create-10k-nft-collection)), it's ineffective. If you want to still use this you will probably have to use the paid plan on either nftport or polygon, I'm not sure which one.

**Comment by giorgioand** [comment]
Hi, have you found anything on this error? i got the same error after 121 nfts. Thanks.

**Comment by Emma071189** [comment]
Hi anybody resolved this?

**Comment by CarlBainbridge** [comment]
My Question @bolshoytoster is

how would one rewrite the code for mint.js to go around nftport so that you could upload to polygon without these limits and without losing the rest of the work put into it before then. 

const fetch = require("node-fetch");
const path = require("path");
const basePath = process.cwd();
const fs = require("fs");

const AUTH = 'nft port api key ';
const CONTRACT_ADDRESS = 'erc721 contract address';
const MINT_TO_ADDRESS = 'your wallet address';
const CHAIN = 'polygon';
const TIMEOUT = 500; // Milliseconds. This a timeout for errors only. If there is an error, it will wait then try again. 5000 = 5 seconds.

if (!fs.existsSync(path.join(`${basePath}/build`, "/minted"))) {
  fs.mkdirSync(path.join(`${basePath}/build`, "minted"));
}

async function main() {
  const ipfsMetas = JSON.parse(
    fs.readFileSync(`${basePath}/build/ipfsMetas/_ipfsMetas.json`)
  );

  for (const meta of ipfsMetas) {
    const mintFile = `${basePath}/build/minted/${meta.custom_fields.edition}.json`;

    try {
      fs.accessSync(mintFile);
      const mintedFile = fs.readFileSync(mintFile)
      if(mintedFile.length > 0) {
        const mintedMeta = JSON.parse(mintedFile)
        if(mintedMeta.mintData.response !== "OK") throw 'not minted'
      }
      console.log(`${meta.name} already minted`);
    } catch(err) {
      try {
        let mintData = await fetchWithRetry(meta)
        const combinedData = {
          metaData: meta,
          mintData: mintData
        }
        writeMintData(meta.custom_fields.edition, combinedData)
        console.log(`Minted: ${meta.name}!`);
      } catch(err) {
        console.log(`Catch: ${err}`)
      }
    }
  }
}

main();

function timer(ms) {
  return new Promise(res => setTimeout(res, ms));
}

async function fetchWithRetry(meta)  {
  await timer(TIMEOUT);
  return new Promise((resolve, reject) => {
    const fetch_retry = (_meta) => {
      let url = "https://api.nftport.xyz/v0/mints/customizable";

      const mintInfo = {
        chain: CHAIN,
        contract_address: CONTRACT_ADDRESS,
        metadata_uri: _meta.metadata_uri,
        mint_to_address: MINT_TO_ADDRESS,
        token_id: _meta.custom_fields.edition,
      };

      let options = {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
          Authorization: AUTH,
        },
        body: JSON.stringify(mintInfo),
      };

      return fetch(url, options).then(async (res) => {
        const status = res.status;

        if(status === 200) {
          return res.json();
        }            
        else {
          console.error(`ERROR STATUS: ${status}`)
          console.log('Retrying')
          await timer(TIMEOUT)
          fetch_retry(_meta)
        }            
      })
      .then(async (json) => {
        if(json.response === "OK"){
          return resolve(json);
        } else {
          console.error(`NOK: ${json.error}`)
          console.log('Retrying')
          await timer(TIMEOUT)
          fetch_retry(_meta)
        }
      })
      .catch(async (error) => {  
        console.error(`CATCH ERROR: ${error}`)  
        console.log('Retrying')    
        await timer(TIMEOUT)    
        fetch_retry(_meta)
      });
    }          
    return fetch_retry(meta);
  });
}

const writeMintData = (_edition, _data) => {
  fs.writeFileSync(`${basePath}/build/minted/${_edition}.json`, JSON.stringify(_data, null, 2));
};

**Comment by bolshoytoster** [comment]
@CarlBainbridge being completely honest, I have no clue. I have 0 experience with things like that.

**Comment by CarlBainbridge** [comment]
Yeah me either. I know it has to exist, but I am having a bugger of a time finding a file that can pull the info from ipfs and upload it to opensea without pretty much starting again.

**Comment by bolshoytoster** [comment]
@CarlBainbridge it looks like the code relies on nftport for minting, it's probably best to find a [different script](https://github.com/search?q=mint%20nft).

You could also try asking on [this](https://github.com/codeSTACKr/create-10k-nft-collection) repo.
